pkg/common-controller/snapshot_controller_base.go

```diff
@@ -244,6 +247,12 @@ func (ctrl *csiSnapshotCommonController) snapshotWorker() {
 		}
 		newSnapshot, err := ctrl.checkAndUpdateSnapshotClass(snapshot)
 		if err == nil {
+			klog.V(5).Infof("deleting snapshot %q", key)
+			ctrl.deleteSnapshot(newSnapshot)
+		} else if errors.IsNotFound(err) {
+			// We should still handle deletion events even if the VolumeSnapshotClass
+			// is not found in the cluster
+			klog.V(5).Infof("snapshot %q has been scheduled for deletion and the snapshotClass has already been removed", key)
```

Comment:
Can you combine these two messages too?

Reply:
Messages have been combined into a single conditional block.